add SQLite Backend (#148)
developmentseed/cogeo-mosaic: docs/advanced/backends.md

with MosaicBackend("dynamodb://us-east-1/amosaic") as mosaic:
    assert isinstance(mosaic, cogeo_mosaic.backends.dynamodb.DynamoDBBackend)

with MosaicBackend("sqlite:///mosaic.db:amosaic") as mosaic:
    assert isinstance(mosaic, cogeo_mosaic.backends.sqlite.SQLiteBackend)

with MosaicBackend("file:///amosaic.json.gz") as mosaic:
    assert isinstance(mosaic, cogeo_mosaic.backends.file.FileBackend)

with MosaicBackend("amosaic.json.gz") as mosaic:
    assert isinstance(mosaic, cogeo_mosaic.backends.file.FileBackend)

# Read-Only
with MosaicBackend("stac+https://my-stac.api/search", {"collections": ["satellite"]}, 10, 12) as mosaic:
    assert isinstance(mosaic, cogeo_mosaic.backends.stac.STACBackend)
```

## STAC Backend

> The STACBackend is a **read-only** backend, meaning it's not used to read or write a file. This backend will POST to the input url looking for STAC items which will then be used to create the mosaicJSON in memory.

Reviewer: ```suggestion
The STACBackend is a **read-only** backend, meaning it can't be used to write a file. This backend will POST to the input url looking for STAC items which will then be used to create the mosaicJSON in memory.
```
